zerver/lib/user_groups.py

```diff
@@ -58,6 +58,32 @@ def access_user_groups_as_potential_subgroups(
     return list(user_groups)
 
 
+def access_user_group_for_setting(
+    user_group_id: int,
+    user_profile: UserProfile,
+    *,
+    setting_name: str,
+    allow_only_system_groups: bool = False,
+    cannot_set_to_internet_group: bool = False,
+) -> UserGroup:
+    user_group = access_user_group_by_id(user_group_id, user_profile, for_read=True)
+
+    if allow_only_system_groups and not user_group.is_system_group:
+        raise JsonableError(
+            _("'{}' setting can only be set to system user groups.").format(setting_name)
```

Comment:
Maybe "'{}' must be a system user group" would be clear enough.

Reply:
Updated